### ISSUE TITLE
"Catch all" exception handling

### ISSUE THREAD

**Issue by chrisfilo** [nipy/nipype#1465]
In several places, we are catching all possible exceptions without writing down any tracebacks or reporting more details on the exception. This makes debugging such code very difficult (because unexpected exceptions are handled silently). Reported by @dclark87.

**Comment by chrisfilo** [comment]
Those are the examples:
https://github.com/nipy/nipype/blob/master/nipype/pipeline/plugins/base.py#L384
https://github.com/nipy/nipype/blob/master/nipype/pipeline/plugins/base.py#L354

**Comment by satra** [comment]
this is very hard to set as you would be catching almost all exceptions, IOError, OSError, RuntimeError and others. also we don't have a test suite for specific types of exceptions since those tests would have to be run on all different plugins. i do think we need to work towards these by collecting the exceptions in crashfiles into the provenance record.

but in general those pieces are reliant on many pieces outside of nipype.

**Comment by chrisfilo** [comment]
I agree we don't need to catch only specific exceptions, but we do need to log the exception that was ignored.

What do you mean by "but in general those pieces are reliant on many pieces outside of nipype."?
